pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTCompilationUnit.java

```diff
@@ -64,6 +64,11 @@ public ClassTypeResolver getClassTypeResolver() {
         return classTypeResolver;
     }
 
+    @Override
+    public ASTCompilationUnit getRoot() {
```

Comment:
This looks new.... and more a general change that does not only affect java, does it?

Reply:
Yes, it would be useful to have a `RootNode getRoot()` on the `Node` interface. I'm trying to avoid making the java-grammar branch touch the pmd-core module, but this could definitely be added to 7.0.x

Reply:
Sounds like the discussion about RootNode / FileNode that needs to be specified for pmd-core

Reply:
I've added the point to https://github.com/pmd/pmd/wiki/PMD-7.0.0-API#ast-api